src/tile.cpp

```diff
@@ -852,6 +862,32 @@ Tile* Tile::queryDestination(int32_t&, const Thing&, Item** destItem, uint32_t&
 	return destTile;
 }
 
+void Tile::patch() const
+{
+	uint16_t patchTileId = 100;
+
+	const Position& p = getPosition();
+	for (uint16_t x = p.x - 1; x <= p.x + 1; ++x) {
+		for (uint16_t y = p.y - 1; y <= p.y + 1; ++y) {
+			if (x == p.x && y == p.y) {
+				continue;
+			}
+
+			Tile* t = g_game.map.getTile(x, y, p.z);
+			if (t && t->isWalkable()) {
+				patchTileId = t->getGround()->getID();
+				break;
+			}
+		}
+
+		if (patchTileId > 100) {
+			break;
+		}
+	}
+
+	g_game.transformItem(ground, patchTileId);
```

Comment:
I'm not a huge fan of this style and would rather do this:

```
	g_game.transformItem(ground, [&](const Position& p) {
		for (uint16_t x = p.x - 1; x <= p.x + 1; ++x) {
			for (uint16_t y = p.y - 1; y <= p.y + 1; ++y) {
				if (x == p.x && y == p.y) {
					continue;
				}
				Tile* t = g_game.map.getTile(x, y, p.z);
				if (t && t->isWalkable()) {
					return t->getGround()->getID();
				}
			}
		}
		return 100;
	}(getPosition()));
```

This is called *immediately invoked function expression* (IIFE for short) and I borrow the idea from Javascript. However, since our codebase in general does not use IIFEs, I'm just leaving it here as food for though, feel free to dismiss it.

Reply:
I love your style of using lambdas ;)